### ISSUE TITLE
Concurrent job executions with task caches cause "cannot_invalidate_during_initialization"

### ISSUE THREAD

**Issue by JohannesRudolph** [concourse/concourse#1850]
# Bug Report

We recently started using task caches for some tasks. We have a few jobs in our pipeline that are allowed to run concurrently (e.g. build and test), however we frequently see this issue at the end of a (otherwise successful) task run: 

```
pq: new row for relation "volumes" violates check constraint "cannot_invalidate_during_initialization"
```

Looks like there's a race condition somewhere?

The issue only occurs sometimes when we have two instances of the job/task running concurrently. We force them to all run on the same worker too (it's a tagged worker). 

The following can also be handy:

* Concourse version: 3.6
* Deployment type (BOSH/Docker/binary): docker
* Infrastructure/IaaS: Meshcloud OpenStack

**Comment by shashwathi** [comment]
@JohannesRudolph : This is fixed in concourse 3.9.2 version. Can you verify if you are still seeing this issue?

**Comment by vito** [comment]
This is fixed by #2031.
